### PR TITLE
chore(broker): Add startup and close processes metrics

### LINKED ISSUE #3913: Add broker steps startup/shutdown metrics
**Is your feature request related to a problem? Please describe.**

It would be helpful when deploying changes to know how they impact startup/shutdown (start up in particular is important for quick failover).

**Describe the solution you'd like**

Add metrics to monitor startup/shutdown time, total and split by steps, so we can see the difference over time.

**Describe alternatives you've considered**

Checking the logs - works OK-ish, but easy to overlook.

### PR DESCRIPTION
**Pull request by aivinog1**
## Description

Implement metrics to measure the start and close process steps.

## Related issues

closes #3913 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.broker.system.monitoring;
+
+import io.prometheus.client.Gauge;
+
+public class ProcessLifetimeMetrics {
```
Comment (Zelldon) [inline broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java:12]: A bit unsure with this name. Maybe we can change it to BrokerStepMetrics or BrokerMetrics? What do you think?

File: broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  public static final String ZEEBE_NAMESPACE = "zeebe";
+  public static final String PROCESS_NAME_LABEL = "processName";
+  public static final String STEP_NAME_LABEL = "stepName";
+  private static final Gauge STARTUP_METRIC =
+      Gauge.build()
+          .namespace(ZEEBE_NAMESPACE)
+          .name("broker_process_startup_metric")
```
Comment (Zelldon) [inline broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java:20]:
```suggestion
          .name("broker_start_step_latency")
```

File: broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+          .labelNames(PROCESS_NAME_LABEL, STEP_NAME_LABEL)
+          .create()
+          .register();
+
+  private static final Gauge CLOSE_METRICS =
+      Gauge.build()
+          .namespace(ZEEBE_NAMESPACE)
+          .name("broker_process_close_metric")
```
Comment (Zelldon) [inline broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java:28]:
```suggestion
          .name("broker_close_step_latency")
```

File: broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+public class ProcessLifetimeMetrics {
+
+  public static final String ZEEBE_NAMESPACE = "zeebe";
+  public static final String PROCESS_NAME_LABEL = "processName";
+  public static final String STEP_NAME_LABEL = "stepName";
+  private static final Gauge STARTUP_METRIC =
+      Gauge.build()
+          .namespace(ZEEBE_NAMESPACE)
+          .name("broker_process_startup_metric")
+          .labelNames(PROCESS_NAME_LABEL, STEP_NAME_LABEL)
```
Comment (Zelldon) [inline broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java:21]: Please add an help to the metric, which describes what it actually contains. Like: "Time for each broker start step to complete."

File: broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    this.processName = processName;
+  }
+
+  /**
+   * Meter the time to start for a single step.
+   *
+   * @param stepName the name of the step
+   * @param startupDuration the step start duration in ms
+   */
+  public void meterStartupTimeForStep(String stepName, long startupDuration) {
```
Comment (Zelldon) [inline broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java:45]: Normally we use observe, could you change that please.

File: broker/src/main/java/io/zeebe/broker/bootstrap/StartProcess.java [hunk 6]
```diff
@@ -90,7 +94,7 @@ static long takeDuration(final CheckedRunnable runner) throws Exception {
     return endTime - startTime;
   }
 
-  private static class StartStep {
+  public static class StartStep {
```
Comment (Zelldon) [inline broker/src/main/java/io/zeebe/broker/bootstrap/StartProcess.java:97]: Why is this necessary?

File: broker/src/main/java/io/zeebe/broker/bootstrap/StartProcess.java [hunk 7]
```diff
@@ -100,11 +104,11 @@ static long takeDuration(final CheckedRunnable runner) throws Exception {
       this.startFunction = startFunction;
     }
 
-    String getName() {
+    public String getName() {
```
Comment (Zelldon) [inline broker/src/main/java/io/zeebe/broker/bootstrap/StartProcess.java:107]: Why is this necessary?

File: broker/src/main/java/io/zeebe/broker/bootstrap/StartProcess.java [hunk 8]
```diff
@@ -100,11 +104,11 @@ static long takeDuration(final CheckedRunnable runner) throws Exception {
...
       return name;
     }
 
-    StartFunction getStartFunction() {
+    public StartFunction getStartFunction() {
```
Comment (Zelldon) [inline broker/src/main/java/io/zeebe/broker/bootstrap/StartProcess.java:111]: Why is this necessary?

File: broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java (hunk 4 above)
Comment (Zelldon) [inline broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java:21]: I think we didn't need the "process name", this is automatically added by Prometheus on which pod this metric was scraped.

File: broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+  public static final String ZEEBE_NAMESPACE = "zeebe";
+  public static final String PROCESS_NAME_LABEL = "processName";
+  public static final String STEP_NAME_LABEL = "stepName";
+  private static final Gauge STARTUP_METRIC =
+      Gauge.build()
+          .namespace(ZEEBE_NAMESPACE)
+          .name("broker_process_startup_metric")
+          .labelNames(PROCESS_NAME_LABEL, STEP_NAME_LABEL)
+          .create()
```
Comment (Zelldon) [inline broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java:22]: Register does already create, so this is not necessary.

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
Hey @aivinog1 

thanks for your ongoing contributions to our project. We really appreciate that! :+1: 
I think in general it looks good, there were just some smaller things. I added some comments please see below.

Greets
Chris

File: broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java (hunk 1 above)
Comment (aivinog1) [reply broker/src/main/java/io/zeebe/broker/system/monitoring/ProcessLifetimeMetrics.java:12]: Hm, I like `BrokerStepMetrics`. It is not too broad as `BrokerMetrics` :)

File: broker/src/main/java/io/zeebe/broker/bootstrap/StartProcess.java (hunk 6 above)
Comment (aivinog1) [reply broker/src/main/java/io/zeebe/broker/bootstrap/StartProcess.java:97]: No, this is a useless change, I'll revert this

File: broker/src/main/java/io/zeebe/broker/bootstrap/StartProcess.java (hunk 7 above)
Comment (aivinog1) [reply broker/src/main/java/io/zeebe/broker/bootstrap/StartProcess.java:107]: No, this is a useless change, I'll revert this

File: broker/src/main/java/io/zeebe/broker/bootstrap/StartProcess.java (hunk 8 above)
Comment (aivinog1) [reply broker/src/main/java/io/zeebe/broker/bootstrap/StartProcess.java:111]: No, this is a useless change, I'll revert this

**Comment by aivinog1** [comment]
@Zelldon I've fixed all your review comments, can you take another look, please? :)

**Review by ChrisKujawa — APPROVED** [review]
Hey @aivinog1 

sorry for the late response was a busy week.

Thanks again for the contribution looks good!

Just a small note regarding squashing and force push, would be cool if you can do separate commits to apply review hints. This makes an second review easier, because we can see directly what has changed afterwards. **After** the review is done you should squash your commits :) 

Thanks again :1st_place_medal:

**Comment by aivinog1** [comment]
@Zelldon Thanks, I will do separate commits to apply review hints, I understand now why it is important :)

**Comment by Zelldon** [comment]
CI had some connectivity issues...
bors r+

**Comment by aivinog1** [comment]
> CI had some connectivity issues...
> bors r+

@Zelldon Well, maybe we should try to do it again? :)

**Comment by aivinog1** [comment]
@Zelldon Hi :) I'll try to rebase on the latest develop branch, maybe this will help :)

**Comment by aivinog1** [comment]
@Zelldon Hi! Can you post a failed part of the build log? Maybe something wrong with my changes :)

**Comment by Zelldon** [comment]
@aivinog1 no need. There is an bug in our system, which causes to fail our builds. https://github.com/zeebe-io/zeebe/pull/5631 this will fix it.

**Comment by aivinog1** [comment]
@Zelldon I've rebased on fresh develop branch with #5631, can you trigger build, please? :)

**Comment by Zelldon** [comment]
Thanks @aivinog1 
Bors r+
